x/connectivity/connectivity.go

```diff
@@ -49,13 +51,29 @@ func (err *TestError) Unwrap() error {
 // TestResolverStreamConnectivity uses the given [transport.StreamEndpoint] to connect to a DNS resolver and resolve the test domain.
 // The context can be used to set a timeout or deadline, or to pass values to the dialer.
 func TestResolverStreamConnectivity(ctx context.Context, resolver transport.StreamEndpoint, testDomain string) (time.Duration, error) {
```

Comment:
As discussed in the meeting, let's make this return the span information, something like:

- Dial
  - start, end timestamp
  - error
- Write
  - start, end timestamp
  - error
- Read
  - start, end timestamp
  - error

Later, we can make users of the API report that in whatever system they want (including otel).

For now, let's remove all the otel logic. We can add to the app later.

Reply:
@fortuna I am implementing this by defining Tracer and Span interfaces such that they will be compatible with otel types.

Reply:
We don’t really need that for now. We need to think about the data processing side. Will a genetic structure make it harder?

Reply:
That's what I am aiming for (i.e. generic span) with the caveat of making interfaces compatible with otel. It is almost done. I need to do some testing and will commit changes shortly. We can use existing report package for collecting the data or developers can use otel if they wish -- connectivity package won't have any dependencies on otel